core/services/pipeline/orm.go

```diff
@@ -0,0 +1,194 @@
+package pipeline
+
+import (
+	"context"
+	"time"
+
+	"github.com/jinzhu/gorm"
+	"github.com/pkg/errors"
+	"gopkg.in/guregu/null.v4"
+
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	ormpkg "github.com/smartcontractkit/chainlink/core/store/orm"
+)
+
+//go:generate mockery --name ORM --output ./mocks/ --case=underscore
+
+type ORM interface {
+	// Runner
+	CreateRun(jobSpecID int32) (int64, error)
+	WithNextUnclaimedTaskRun(f func(ptRun TaskRun, predecessors []TaskRun) Result) error
+	AwaitRun(ctx context.Context, runID int64) error
+	NotifyCompletion(pipelineRunID int64) error
+	ResultsForRun(runID int64) ([]Result, error)
+
+	// BridgeTask
+	FindBridge(name models.TaskType) (models.BridgeType, error)
+}
+
+type orm struct {
+	db *gorm.DB
+}
+
+var _ ORM = (*orm)(nil)
+
+func NewORM(db *gorm.DB) *orm {
+	return &orm{db}
+}
+
+// CreateRun adds a Run record to the DB, and one TaskRun
+// per TaskSpec associated with the given Spec.  Processing of the
+// TaskRuns is maximally parallelized across all of the Chainlink nodes in the
+// cluster.
+func (o *orm) CreateRun(jobSpecID int32) (int64, error) {
+	var runID int64
+	err := ormpkg.GormTransaction(o.db, func(tx *gorm.DB) error {
+		// Fetch the spec ID
+		var spec Spec
+		err := o.db.
```

Comment:
This fetch feels unnecessary?

Reply:
How else could we do it

Reply:
Do what? The foreign key will automatically verify that the job spec exists in the following statement